python/dgl/nn/pytorch/explain/pgexplainer.py

```diff
@@ -0,0 +1,351 @@
+        graph : DGLGraph
+            A homogeneous graph.
+        feat : Tensor
+            The input feature of shape :math:`(N, D)`. :math:`N` is the
+            number of nodes, and :math:`D` is the feature size.
+        tmp : float
+            The temperature parameter fed to the sampling procedure.
+
+        Returns
+        -------
+        Tensor
+            A scalar tensor representing the loss.
+        """
+        self.model = self.model.to(graph.device)
+        self.elayers = self.elayers.to(graph.device)
+
+        pred = self.model(graph, feat).argmax(-1).data
+
+        prob, edge_mask = self.explain_graph(
+            graph, feat, tmp=tmp, training=True
+        )
+
+        self.edge_mask = edge_mask
+
+        loss_tmp = self.loss(prob.unsqueeze(dim=0), pred)
+        return loss_tmp
+
+    def explain_graph(self, graph, feat, tmp=1.0, training=False):
+        r"""Learn and return an edge mask that plays a crucial role to
+        explain the prediction made by the GNN for a graph. Also, return
+        the prediction made with the edges chosen based on the edge mask.
+
+        Parameters
+        ----------
+        graph : DGLGraph
+            A homogeneous graph.
+        feat : Tensor
+            The input feature of shape :math:`(N, D)`. :math:`N` is the
+            number of nodes, and :math:`D` is the feature size.
+        tmp : float
+            The temperature parameter fed to the sampling procedure.
+        training : bool
+            We indicate we want to train the explanation network if
+            set to True. If set to False, we indicate that we want
+            to find the edges of the graph that contribute most to
+            the graph explanations.
+
+        Returns
+        -------
+        Tensor, Tensor
+            The classification probability for graph with edge mask,
+            the edge weights where higher edge weights indicate that
+            they contribute to the explanation.
+
+        Examples
+        --------
+
+        >>> import torch as th
+        >>> import torch.nn as nn
+        >>> import dgl
+        >>> from dgl.data import GINDataset
+        >>> from dgl.dataloading import GraphDataLoader
+        >>> from dgl.nn import GraphConv, PGExplainer
+        >>> import numpy as np
+
+        >>> # Define the model
+        >>> class Model(nn.Module):
+        ...     def __init__(self, in_feats, out_feats):
+        ...         super().__init__()
+        ...         self.conv = GraphConv(in_feats, out_feats)
+        ...         self.fc = nn.Linear(out_feats, 1)
+        ...         nn.init.xavier_uniform_(self.fc.weight)
+        ...
+        ...     def forward(self, g, h, embed=False, edge_weight=None):
+        ...         h = self.conv(g, h, edge_weight=edge_weight)
+        ...         if not embed:
+        ...             g.ndata['h'] = h
+        ...             hg = dgl.mean_nodes(g, 'h')
+        ...             return th.sigmoid(self.fc(hg))
```

Comment:
From your implementation, it seems that this function should return the logits rather than the probabilities.

In addition, since this module is supposed to only support multiclass classification, the model should output two values corresponding to the two classes rather than one value.

Reply:
The model should return probability, as we assume it to be a standard multi-class classification model. I have updated how the loss was calculated for the pgexplainer. Please take a look and let me know.